dropwizard-configuration/src/test/java/io/dropwizard/configuration/SubstitutingSourceProviderTest.java

```diff
@@ -21,10 +23,19 @@ public String lookup(String key) {
                 return "baz";
             }
         };
-        SubstitutingSourceProvider provider = new SubstitutingSourceProvider(new DummySourceProvider(), new StrSubstitutor(dummyLookup));
+        DummySourceProvider dummyProvider = new DummySourceProvider();
+        SubstitutingSourceProvider provider = new SubstitutingSourceProvider(dummyProvider, new StrSubstitutor(dummyLookup));
         String results = new String(ByteStreams.toByteArray(provider.open("foo: ${bar}")), StandardCharsets.UTF_8);
 
         assertThat(results).isEqualTo("foo: baz");
+
+        // ensure that opened streams are closed
+        try {
+            dummyProvider.lastStream.read();
+            fail("IOException expected; stream should be closed");
```

Comment:
Could we change this to `failBecauseExceptionWasNotThrown(IOException.class);` instead?